flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/reconciler/deployment/ApplicationReconciler.java

```diff
@@ -318,6 +319,31 @@ private boolean shouldRestartJobBecauseUnhealthy(
         return restartNeeded;
     }
 
+    private boolean cleanupTerminalJmAfterTtl(
+            FlinkDeployment deployment, Configuration observeConfig) {
+        var status = deployment.getStatus();
+        boolean terminal = ReconciliationUtils.isJobInTerminalState(status);
+        boolean jmStillRunning =
+                status.getJobManagerDeploymentStatus() != JobManagerDeploymentStatus.MISSING;
+
+        if (terminal && jmStillRunning) {
+            var ttl = observeConfig.get(KubernetesOperatorConfigOptions.OPERATOR_JM_SHUTDOWN_TTL);
+            boolean ttlPassed =
+                    Instant.now()
```

Comment:
I know that the window in the test is big enough but using `now` in tests may make it flaky. I would add `Clock` param either to the function or as member var. In tests it can be `Clock.fixed...` and in PROD code `Clock.systemDefaultZone()`.